webpack/assets/javascripts/react_app/components/HostDetails/Tabs/Details/Cards/Virtualization/index.js

```diff
@@ -0,0 +1,75 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {
+  DescriptionList,
+  DescriptionListTerm,
+  DescriptionListGroup,
+  DescriptionListDescription,
+} from '@patternfly/react-core';
+import { translate as __ } from '../../../../../../common/I18n';
+import { foremanUrl } from '../../../../../../common/helpers';
+import { useAPI } from '../../../../../../common/hooks/API/APIHooks';
+import CardTemplate from '../../../../Templates/CardItem/CardTemplate';
+import VirtVmware from './VirtVmware';
+import VirtOvirt from './VirtOvirt';
+import VirtLibvirt from './VirtLibvirt';
+import VirtEc2 from './VirtEc2';
+import VirtGce from './VirtGce';
+import VirtOpenstack from './VirtOpenstack';
+
+const VirtualizationCard = ({ hostDetails, hostName }) => {
+  const {
+    compute_resource_id: computeResourceId,
+    compute_resource_name: computeResourceName,
+    compute_resource_provider: provider,
+  } = hostDetails;
+  const virtUrl = foremanUrl(`/api/hosts/${hostName}/vm_compute_attributes`);
+  const { response: vm, status } = useAPI('get', virtUrl);
+
+  if (!provider) return null;
+  if (status !== 'RESOLVED') return null;
+
+  const components = {
+    vmware: VirtVmware,
+    libvirt: VirtLibvirt,
+    ovirt: VirtOvirt,
+    ec2: VirtEc2,
+    gce: VirtGce,
+    openstack: VirtOpenstack,
+  };
+  const VirtCardDetails = components[provider];
```

Comment:
What happens if the compute resource is not found? Shouldn't it `return null` if `VirtCardDetails` is null or handle it some other way? Remember that plugins can also add compute resources. I understand you may not want to make this plugin aware, but it shouldn't crash.

Reply:
Nice catch 👍🏻